### PR TITLE
(PC-4934) New password : Positioning Bottom card along with keyboard …

### PR DESCRIPTION
**Pull request by blixit**
…&& Fixing Android deeplink listener

Link to JIRA ticket: https://passculture.atlassian.net/browse/PC-XXX

## Checklist

I have:

- [x] Made sure the title of my PR follows the convention `[PC-XXX] <summary>`.
- [x] Made sure my feature is working on the relevant real / virtual devices.
- [x] Written **unit tests** for my feature.
- [x] Added a **screenshot** for UI tickets.
- [ ] Explained my technical strategy below if feature is complex.

image

## Deploy hard

If native code (ios/android) was modified, **after** the PR is merged, on the master branch, upgrade the **app version** (+1 patch):

- if you want an hard deployment of the testing environment, use `yarn version:testing` (this will create a commit with a tag)
- then run `git push --follow-tags`

## Technical strategy

> _Insert technical strategy_

### CODE REVIEW

File: src/features/deeplinks/listener.android.ts [hunk 1]
```diff
@@ -20,5 +20,7 @@ export function useListenDeepLinksEffect() {
         }
       })
       .catch(() => onError(_(t`Impossible d'ouvrir le lien`)))
+    Linking.addEventListener('url', handleDeeplinkUrl)
+    return () => Linking.removeEventListener('url', handleDeeplinkUrl)
```
Comment (BenStirrup) [inline src/features/deeplinks/listener.android.ts:24]: C'est lié au clavier à la position du clavier ce code là ? ^^

File: src/ui/components/keyboard/index.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import { Platform } from 'react-native'
+
+import { AvoidingKeyboardContainer as AvoidingKeyboardContainerAndroid } from './AvoidingKeyboardContainer.android'
+import { AvoidingKeyboardContainer as AvoidingKeyboardContainerIos } from './AvoidingKeyboardContainer.ios'
+
+export const AvoidingKeyboardContainer =
+  Platform.OS === 'ios' ? AvoidingKeyboardContainerIos : AvoidingKeyboardContainerAndroid
```
Comment (BenStirrup) [inline src/ui/components/keyboard/index.ts:7]: Pourquoi tu utilises `Platform.OS` alors que les extensions de fichier font déjà ce job ?

File: src/features/deeplinks/listener.android.ts (hunk 1 above)
Comment (blixit) [reply src/features/deeplinks/listener.android.ts:24]: oui en partie sur Android !

File: src/ui/components/keyboard/index.ts (hunk 2 above)
Comment (blixit) [reply src/ui/components/keyboard/index.ts]: bien vu ! done !

**Review by BenStirrup — APPROVED** [review]

**Pull request merged**